fuzz/src/erc20.rs

```diff
@@ -0,0 +1,264 @@
+#![cfg(feature = "solc-backend")]
+use fe_compiler_test_utils::*;
+use serde::{Deserialize, Serialize};
+// Create fuzz test for the contracts
+
+const ALICE: &str = DEFAULT_CALLER;
+const BOB: &str = "2000000000000000000000000000000000000002";
+const JAMES: &str = "3000000000000000000000000000000000000003";
+const TOTAL_SUPPLY: &str = "1000000000000000000000000";
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Copy)]
+pub struct TokenDetails<'a> {
+    pub name: &'a str,
+    pub symbol: &'a str,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Copy)]
+pub struct TransferInfo<'a> {
+    pub to: &'a str,
+    pub value: u128,
+}
+
+pub fn fe_erc20_constructor(details: TokenDetails) {
+    if !details.name.is_empty() {
+        with_executor(&|mut executor| {
+            let harness = deploy_contract(
+                &mut executor,
+                "test-utils/fuzz/fixtures/fe/erc20.fe", // TODO: Need to fix the path
+                "ERC20",
+                &[string_token(details.name), string_token(details.symbol)],
+            );
+            erc20_constructor(details, &mut executor, harness);
+        });
+    }
+}
+
+pub fn sol_erc20_constructor(details: TokenDetails) {
+    if !details.name.is_empty() {
+        with_executor(&|mut executor| {
+            let harness = deploy_contract(
+                &mut executor,
+                "fuzz/fixtures/solidity/erc20.sol",
+                "ERC20",
+                &[string_token(details.name), string_token(details.symbol)],
+            );
+            erc20_constructor(details, &mut executor, harness);
+        });
+    }
+}
+
+pub fn fe_erc20_transfer(info: TransferInfo) {
+    if info.value >= 1000000000000000000000000 as u128 {
+        with_executor(&|mut executor| {
+            let mut harness = deploy_contract(
+                &mut executor,
+                "test-utils/fuzz/fixtures/fe/erc20.fe", // TODO: Need to fix the path
+                "ERC20",
+                &[string_token("Fe Coin"), string_token("FE")],
```

Comment:
This is a good start, but it doesn't test much.

I think what we eventually want is to have two separate deployed contracts/executors that we perform many state-modifying transactions on in a random order. This will get us into edge cases where we might find bugs.

Of course this doesn't need to be included in this PR, I'm just wondering if you've been thinking about this.

Reply:
Yeah, I was thinking about this to have some kind of random order testing. I have a process in mind where we can come up with some mutation testing where we can randomly mutate the contract as well if we want but that is I think one step forward but for now, I am thinking to have some kind of prop testing with random ordering.